Improve TextField API and styles

viktor-yakubiv opened the pull request:
Introduces class-first approach for the inputs.

Fixes the bug with the Firefox broken label animation by removing the `-webkit-autofill` selector.

---

@Joozty I wanted to fix the bug and ended up rewriting everything. Even if it seems not worth it, it is for me. This may be the first step of writing the next forms API in the library. Let me know what do you think about.

Joozty commented on src/elements/forms/text-field.css:
```diff
@@ -77,52 +137,49 @@ label {
+.pure {
+  /* Kepping screenreader-only tweak as is (with !important-s) for a safety */
+  /* stylelint-disable declaration-no-important */
+  & > .label {
```
This is duplicate code. What about applying sr-only class on JS level? :)

viktor-yakubiv commented:
I fixed Stylint issues and rebased to solve conflicts. I hope I did not break anything and wonder what changes could happen with the text field today.

Joozty commented:
@viktor-yakubiv just some empty lines were removed from there.

Joozty commented:
As we already talk about it the autofill is crucial here and it should stay in there because otherwise chrome is broken. Apart from that refactor is wonderful. :)
![pass](https://user-images.githubusercontent.com/12775440/84676873-43daf180-af2e-11ea-9684-e4abfd6777e7.png)

viktor-yakubiv commented:
@Joozty I tried to return autofill but the only thing I know that it does not break Firefox now. Could you please check it?

I decided to duplicate the code for now and use the mixins later when we add tooling for it.

viktor-yakubiv commented:
@Joozty I am only not sure about having success/error for a 'pure' TextField. Currently, it's not possible but it should be possible probably. We have to distinguish layout and colour variants but I am struggling with consistency vs new API decisions.
